File: src/main/java/hilfling/backend/hilfing/model/PhotoOnPurchaseOrder.java
package hilfling.backend.hilfing.model;

import lombok.Data;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity(name = "photo_on_purchase_orders")
@Table(name = "photo_on_purchase_orders", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"purchase_order_id", "photo_id", "size"})
})
public class PhotoOnPurchaseOrder implements Serializable, BaseModel {
    public PhotoOnPurchaseOrder() {
    }

    public PhotoOnPurchaseOrder(PurchaseOrder purchaseOrder, Photo photo, String size, Integer amount) {
        this.purchaseOrder = purchaseOrder;
        this.photo = photo;
        this.size = size;
        this.amount = amount;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "purchase_order_id", referencedColumnName = "id", nullable = false)
    private PurchaseOrder purchaseOrder;

    @ManyToOne
    @JoinColumn(name = "photo_id", referencedColumnName = "id", nullable = false)
    private Photo photo;

    @Column(name = "size", nullable = false)
    private String size;

    @Column(name = "amount", nullable = false)
    @ColumnDefault("1")
    private Integer amount;
}

File: src/main/kotlin/hilfling/backend/hilfling/model/PhotoOnPurchaseOrder.kt
package hilfling.backend.hilfing.model

import me.liuwj.ktorm.database.Database
import me.liuwj.ktorm.entity.Entity
import me.liuwj.ktorm.entity.sequenceOf
import me.liuwj.ktorm.schema.*

interface PhotoOnPurchaseOrder : BaseModel<PhotoOnPurchaseOrder> {
    companion object : Entity.Factory<PhotoOnPurchaseOrder>()

    var size: String
    var amount: Int

    // Foreign keys
    var purchaseOrder: PurchaseOrder
    var photo: Photo
}

object PhotoOnPurchaseOrders : BaseTable<PhotoOnPurchaseOrder>("photo_on_purchase_order") {
    val size = varchar("size").bindTo { it.size }
    val amount = int("amount").bindTo { it.amount }

    // Foreign keys
    val purchaseOrderId = int("purchase_order_id").references(PurchaseOrders){it.purchaseOrder}
    val photoId = int("photo_id").references(Photos){it.photo}
}

val Database.photo_on_purchase_orders get() = this.sequenceOf(PhotoOnPurchaseOrders)